tools/commit_slider/case_1.md

```diff
@@ -0,0 +1,316 @@
+
+    # passing each of the commandlines for execution
+    simpleSubstitute(cfg, "actualAppCmd", "$.runConfig.firstAppCmd", "$.appCmd")
+
+    # run first app
+    checkOut = fetchAppOutput(cfg, commit)
+    commitLogger.info(checkOut)
+    foundThroughput = re.search(
+        self.outPattern, checkOut, flags=re.MULTILINE
+    ).group(1)
+
+    # parsing of output and holding it with full commandline and specific model for the final result
+    self.firstThroughput = foundThroughput
+    self.firstModel = cfg['appCmd']
+    fullOutput = checkOut
+    simpleSubstitute(cfg, "actualAppCmd", "$.runConfig.secondAppCmd", "$.appCmd")
+
+    # run second app
+    checkOut = fetchAppOutput(cfg, commit)
+    commitLogger.info(checkOut)
+    foundThroughput = re.search(
+        self.outPattern, checkOut, flags=re.MULTILINE
+    ).group(1)
+    self.secondThroughput = foundThroughput
+    self.secondModel = cfg['appCmd']
+
+    # add commit data to commit path
+    pc = Mode.CommitPath.PathCommit(
+        commit,
+        Mode.CommitPath.CommitState.DEFAULT
+    )
+    self.setOutputInfo(pc)
+    self.commitPath.accept(self.traversal, pc)
+
+def  printResult(self):
+    # formatting of full output
+    for  pathcommit  in  self.commitPath.getList():
+        commitInfo = self.getCommitInfo(pathcommit)
+        print(commitInfo)
+        self.outLogger.info(commitInfo)
+
+def  setOutputInfo(self, pathCommit):
+    # populating of commit path
+    pathCommit.firstThroughput = self.firstThroughput
+    pathCommit.firstModel = self.firstModel
+    pathCommit.secondThroughput = self.secondThroughput
+    pathCommit.secondModel = self.secondModel
+
+def  getCommitInfo(self, commit):
+    return  "{hash}, throughput_1 = {t1}, throughput_2 = {t2}".format(
+        hash=commit.cHash,
+        t1=commit.firstThroughput,
+        t2=commit.secondThroughput)
+```
+### Configuration
+```
+supposedly we pass both commandlines launching models in two parameters cfg["runConfig"]["firstAppCmd"] and cfg["runConfig"]["secondAppCmd"], i.e.
+cfg = { 
+    "appCmd" : "{actualAppCmd}",
+    "appPath": "<path_to_benchmark_app>",
+    "gitPath" : "<path_to_git_repository>",
+    "buildPath" : "<path_to_build_dir>",
+        "runConfig" : {
+            "commitList" : {
+                "explicitList" : [ "{start}", "{end}" ]
+            },
+            "mode" : "crossCheck",
+            "firstAppCmd" : "./benchmark_app -m model_1.xml --other_params",
+            "secondAppCmd" : "./benchmark_app -m model_2.xml --other_params",
+            "traversal" : "bruteForce"
+        }
+}
+```
+### Output
+```
+python3 commit_slider.py -cfg cfg.json
+
+<hash_1>, throughput_1 = 500.0, throughput_2 = 500.0
+<hash_2>, throughput_1 = 1000.0, throughput_2 = 1000.0
+
+```
+We've got rid of full output and learned to provide only necessary information, but we still limited by verbose configuration-style API and still provide raw output information.
+
+## Third Version: Specific Template
```

Comment:
Minor: please use a common style for Version naming. I'd suggest "Version 1, Version 2, Version 3"

Reply:
Applied